Get SchwiftyTests/CompleterTests.swift

```diff
@@ -0,0 +1,2477 @@
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "atest"
+        let expectedRange = NSMakeRange(1, 0) // After the "a"
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_AtTheEnd_InsertedNormally() {
+        let text = "test"
+        let range = NSMakeRange(4, 0) // End
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "testa"
+        let expectedRange = NSMakeRange(5, 0) // End
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_InTheMiddleOfWhitespace_InsertedNormally() {
+        let text = "tes\t  \tt"
+        let range = NSMakeRange(5, 0) // After the first space
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "tes\t a \tt"
+        let expectedRange = NSMakeRange(6, 0) // After the "a"
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_BetweenTabs_InsertedNormally() {
+        let text = "tes\t\tt"
+        let range = NSMakeRange(4, 0) // Between the tabs
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "tes\ta\tt"
+        let expectedRange = NSMakeRange(5, 0) // After the "a"
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_BetweenSpaces_InsertedNormally() {
+        let text = "tes  t"
+        let range = NSMakeRange(4, 0) // After the first space
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "tes a t"
+        let expectedRange = NSMakeRange(5, 0) // After the "a"
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_WithEmojis_InsertedNormally() {
+        let text = "👱🏾🚀test"
```

Comment:
Not fully following this convo, but to test whether you're doing Unicode correctly, be sure to test with some of the following:

1. A country flag. Those are actually two distinct characters ("C" and "A" for Canada, but using a special country flag emoji alphabet)
2. An Emoji with a skin tone modifier (I think that one was a surrogate pair)
3. Any other "regular Emoji" (those are single code points IIRC).

It's been a while since I had to muck with Unicode at the byte level, but definitely be aware of multiple "characters" for one character (even in UTF-32), and the case of both ones that can be detected as being one character (like combining accents on un-accented letters) and ones that are two distinct characters but produce a single glyph (like country flags).